src/screens/Connections/Connections.tsx

```diff
@@ -71,6 +75,16 @@ const Connections = () => {
     }, 2000);
   }, []);
 
+  const onPress = () => {
+    if (modalRef && modalRef.current && modalRef.current.close) {
+      modalRef.current.close();
+    }
+
+    navigation.navigate('BookingScreen', {
+      modalData: modalData,
+    });
```

Comment:
(This is a silly one, but I'm used to ESLint reminding it to me 🤭)
If you'd like, this could be simplified: since the value name and the prop key are the same, it can be sent like
```suggestion
    navigation.navigate('BookingScreen', {
      modalData,
    });
```
Again, this is not a super important change (it'll work and it's still readable), it's more style-wise/consistency and depends on what styling rules we decide to follow!